### ISSUE TITLE
Add all the files under `threejs/examples/jsm/libs` to `three-stdlib/src/libs`

### ISSUE THREAD

**Issue by shuta13** [pmndrs/three-stdlib#77]
### Describe the feature you'd like:

I would like to be able to manage all files in `three.js/examples/jsm/libs` with `three-stdlib`.

For now, only `meshopt_decoder.module.js` is copied from `three.js/examples/jsm/libs` to `three-stdlib` as `MeshoptDecorder.js` ( https://github.com/pmndrs/three-stdlib/blob/main/src/libs/MeshoptDecoder.js ), but other files should be moved to `three-stdlib` as well. By moving other files to `three-stdlib` and rewriting them into TypeScript if possible, I think `three-stdlib` can better support people using `three.js/examples`.

### Suggested implementation:

Follow the steps below

- copy from `three.js/examples/jsm/libs` to `three-stdlib/src/libs`
- move `three-stdlib/src/libs/*.module.js` into `three-stdlib/src/libs/*.js`
- (if possible) move `three-stdlib/src/libs/*.js` into `three-stdlib/src/libs/*.ts`

**Comment by looeee** [comment]
Would it be better to just link to the respective NPM libraries? Maintaining separate copies of third-party dependencies is quite a rabbit hole.

**Comment by shuta13** [comment]
@looeee 

> Would it be better to just link to the respective NPM libraries? Maintaining separate copies of third-party dependencies is quite a rabbit hole.

There are other ways to do this, but for now, three.js manages copies from each library under `threejs/examples/jsm/libs`. The intention is to move them to three-stdlib and reduce the maintenance load on the three.js side.

**Comment by drcmda** [comment]
yes, i have tried to use the npm packages for all libs, the only one that does not have a package out there is meshopt. imo that allows us to have better semantic safety, we can know when something breaks or if there is a patch update.

**Comment by looeee** [comment]
> three.js manages copies from each third-party library under threejs/examples/jsm/libs

That at least started for historical reasons when many of the libs (an three.js iteself) weren't on NPM.

**Comment by shuta13** [comment]
@drcmda 

> yes, i have tried to use the npm packages for all libs, the only one that does not have a package out there is meshopt sadly. i think the reason that three did not do this is because it does not want to be a npm module, similar to how they initially did not want to import from "three" in the jsm classes. but imo real imports allow us to have better semantic safety, we can know when some of these libraries have breaking changes or if there is a patch update. managing them by hand seems like a lot of trouble.

I see, so this library had such a past... Also, I think that what you are saying is true.

> btw @shuta13 you're doing a lot of good work on this library, you want in? we definitively could need more eyes and hands.

Thank you very much. If there is anything I can do to help, I would love to participate and work on it.

**Comment by shuta13** [comment]
@drcmda 

I'm sorry to bother you. I have not heard from you for a while, so I will send you a message again.

Is it correct that it is difficult to migrate all `threejs/examples/jsm/libs` to `three-stdlib` at the moment?
Also, will it be possible for me to join your community and help maintain this library and others in the near future?

**Comment by joshuaellis** [comment]
>Also, will it be possible for me to join your community and help maintain this library and others in the near future?

I will add you now.

>Is it correct that it is difficult to migrate all threejs/examples/jsm/libs to three-stdlib at the moment?

we're using the npm modules. Is there a reason you think we need to add the actual libs?
